scan/pki.go

```diff
@@ -88,6 +89,15 @@ func chainValidation(host string) (grade Grade, output Output, err error) {
 		return
 	}
 
+	h, _, err := net.SplitHostPort(host)
+	if err != nil {
+		return
+	}
+
+	if err = chain[0].VerifyHostname(h); err != nil {
```

Comment:
Wait, there should be a check that the server sends us at least one certificate...